userspace/libsinsp/test/sinsp_metrics.ut.cpp

```diff
@@ -0,0 +1,382 @@
+		} else if (strncmp(metric.name, "duration_sec", 13) == 0)
+		{
+			prometheus_text_substring = R"(# HELP testns_falco_duration_seconds_total https://falco.org/docs/metrics/
+# TYPE testns_falco_duration_seconds_total counter
+testns_falco_duration_seconds_total{raw_name="duration_sec"} 144
+)";
+			ASSERT_TRUE(prometheus_text.find(prometheus_text_substring) != std::string::npos) << "Substring not found in prometheus_text";
+		} else if (strncmp(metric.name, "evt_rate_sec", 13) == 0)
+		{
+			prometheus_text_substring = R"(# HELP testns_falco_evt_rate_seconds https://falco.org/docs/metrics/
+# TYPE testns_falco_evt_rate_seconds gauge
+testns_falco_evt_rate_seconds{raw_name="evt_rate_sec"} 126065.400000
+)";
+			ASSERT_TRUE(prometheus_text.find(prometheus_text_substring) != std::string::npos) << "Substring not found in prometheus_text";
+		} else if (strncmp(metric.name, "host_boot_ts", 13) == 0)
+		{
+			prometheus_text_substring = R"(# HELP testns_falco_host_boot_timestamp_nanoseconds https://falco.org/docs/metrics/
+# TYPE testns_falco_host_boot_timestamp_nanoseconds gauge
+testns_falco_host_boot_timestamp_nanoseconds{raw_name="host_boot_ts"} 1708753667000000000
+)";
+			ASSERT_TRUE(prometheus_text.find(prometheus_text_substring) != std::string::npos) << "Substring not found in prometheus_text";
+		}
+	}
+
+}
+
+TEST_F(sinsp_with_test_input, sinsp_libs_metrics_collector_output_rule)
+{
+	m_inspector.set_sinsp_stats_v2_enabled();
+	// Extra call to verify that we don't fail
+	m_inspector.set_sinsp_stats_v2_enabled();
+	DEFAULT_TREE
+	auto evt = generate_random_event(p2_t1_tid);
+	ASSERT_EQ(get_field_as_string(evt, "proc.nthreads"), "3");
+
+	/* Snapshot current metrics and get the updated metrics_snapshot buffer */
+	uint32_t test_metrics_flags = (METRICS_V2_KERNEL_COUNTERS | METRICS_V2_LIBBPF_STATS | METRICS_V2_RESOURCE_UTILIZATION | METRICS_V2_STATE_COUNTERS);
+	auto libs_metrics_collector = std::make_unique<libs::metrics::libs_metrics_collector>(&m_inspector, test_metrics_flags);
+	auto output_rule_metrics_converter = std::make_unique<libs::metrics::output_rule_metrics_converter>();
+
+	/* Multiple calls */
+	libs_metrics_collector->snapshot();
+	auto metrics_snapshot = libs_metrics_collector->get_metrics();
+	libs_metrics_collector->snapshot();
+	libs_metrics_collector->snapshot();
+	metrics_snapshot = libs_metrics_collector->get_metrics();
+	ASSERT_EQ(metrics_snapshot.size(), 28);
+
+	/* These names should always be available, note that we currently can't check for the merged scap stats metrics here */
+	std::unordered_set<std::string> minimal_metrics_names = {"cpu_usage_perc", "memory_rss_kb", "host_open_fds", \
+	"n_threads", "n_fds", "n_added_fds", "n_added_threads", "n_removed_threads", "n_containers"};
+
+	for(const auto& metric_name : minimal_metrics_names)
+	{
+		uint32_t i = 0;
+		for (const auto& metric: metrics_snapshot) 
+		{
+			if(metric_name.compare(metric.name) == 0)
```

Comment:
Nit: I think there's an overload of `std::string::operator==()` that takes a `const char*`.